### PR TITLE
ci: Split extension integration tests

### PR DESCRIPTION
**Pull request by olix0r**
The viz tests are especially flakey, and there's relatively little value
to running them on changes that don't touch viz components (or the
proxy).

This change splits out integration tests so that the main integration
test skips building extension components and only runs tests that use
extensions when necessary.

Signed-off-by: Oliver Gould <ver@buoyant.io>

### CODE REVIEW

File: .github/workflows/integration-ext.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+          name: image-archives
+          path: /home/runner/archives
+
+  test:
+    needs: [build]
+    strategy:
+      matrix:
+        test:
+          - cluster-domain
+          - default-policy-deny
```
Comment (adleong) [inline .github/workflows/integration-ext.yml:61]: Does this mean that these integration tests which test core functionality but rely on extensions to do (like viz) are only run if a file in an extension directory changes?  That's unfortunate.

Comment (adleong) [reply .github/workflows/integration-ext.yml:61]: Should these tests of core functionality be in `integration.yml` instead?  And just leave the tests which specifically test extensions here?

Comment (olix0r) [reply .github/workflows/integration-ext.yml:61]: I'd really like to not have to build all of viz/web/etc, deploy prometheus, etc to test core functionality. I would much rather split up the install test to stop doing anything with viz.

**Comment by olix0r** [comment]
The good stuff is inhttps://github.com/linkerd/linkerd2/pull/8964 -- we can revisit conditionally running viz tests...

**Pull request closed without merging**